deepracer/edit_action_speeds/editActionSpeeds.py

```diff
@@ -0,0 +1,61 @@
+import json
+import os
+import uuid
+
+filename = 'action_space.json'
+
+# README
+# Paste in your full action space into the file "action_space.json", it can handle index values if you have them already
+# Run this script with desired parameters on the last line"
+
+
+def edit_action_speeds(edit_value, change_left_steering, change_straight, change_right_steering, add_indexer):
+    with open(filename) as action_space:
+        data = json.load(action_space)
+        index_counter = 0
+        for action in data['action_space']:
+            if add_indexer:
+                action['index'] = index_counter
+                index_counter += 1
+            if change_left_steering:
+                if action['steering_angle'] < 0:
+                    action_update(action, edit_value)
+            if change_straight:
+                if action['steering_angle'] == 0:
+                    action_update(action, edit_value)
+            if change_right_steering:
+                if action['steering_angle'] > 0:
+                    action_update(action, edit_value)
+
+    # Create randomly named temporary file to avoid interference with other thread/asynchronous request
+    tempfile = os.path.join(os.path.dirname(filename), str(uuid.uuid4()))
```

Comment:
Python has a library for temporary files :)
https://docs.python.org/3/library/tempfile.html

Reply:
Spent nearly 2 hours trying tempfile.TemporaryFile() and showed you the results, for the errors it gives and the change in 1 line I don't think it's worth pushing more on.